utils/DeployUtils.ps1

```diff
@@ -0,0 +1,240 @@
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $GalleryName = 'PrivateScubaGearGallery',
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyString()]
+        [string]
+        $OverrideModuleVersion = ""
+    )
+
+    $ModuleBuildPath = Build-ScubaModule -ModulePath $ModulePath -OverrideModuleVersion $OverrideModuleVersion
+
+    if (SignScubaGearModule -ModulePath $ModuleBuildPath){
+        Publish-Module -Path $ModuleBuildPath -Repository $GalleryName
+    }
+    else {
+        Write-Error "Failed to sign module."
+    }
+}
+
+function Build-ScubaModule{
+    <#
+        .NOTES
+        Internal helper function
+    #>
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Container})]
+        [string]
+        $ModulePath,
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyString()]
+        [string]
+        $OverrideModuleVersion = ""
+    )
+    $Leaf = Split-Path -Path $ModulePath -Leaf
+    $ModuleBuildPath = Join-Path -Path $env:TEMP -ChildPath $Leaf
+
+    if (Test-Path -Path $ModuleBuildPath -PathType Container){
+        Remove-Item -Recurse -Force $ModuleBuildPath
+    }
+
+    Copy-Item $ModulePath -Destination $env:TEMP -Recurse
+    if (-not (ConfigureScubaGearModule -ModulePath $ModuleBuildPath -OverrideModuleVersion $OverrideModuleVersion)){
+        Write-Error "Failed to configure scuba module for publishing."
+    }
+
+    return $ModuleBuildPath
+}
+
+function ConfigureScubaGearModule{
+    <#
+        .NOTES
+        Internal helper function
+    #>
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Container})]
+        $ModulePath,
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyString()]
+        [string]
+        $OverrideModuleVersion = ""
+    )
+    #TODO: Add any module configuration needed (e.g., adjust Module Version)
+
+    $ManifestPath = Join-Path -Path $ModulePath -ChildPath "ScubaGear.psd1"
+    $ModuleVersion = $OverrideModuleVersion
+
+    if ([string]::IsNullOrEmpty($OverrideModuleVersion)){
+        $CurrentModuleVersion = (Import-PowerShellDataFile $ManifestPath).ModuleVersion
+        $TimeStamp = [int32](Get-Date -UFormat %s)
+        $ModuleVersion = "$CurrentModuleVersion.$TimeStamp"
+    }
+
+    $ManifestUpdates = @{
+        Path = $ManifestPath
+        ModuleVersion = $ModuleVersion
+        ProjectUri = "https://github.com/cisagov/ScubaGear"
+        LicenseUri = "https://github.com/cisagov/ScubaGear/blob/main/LICENSE"
+        Tags = 'CISA', 'AzureAD', 'Cloud', 'Configuration', 'Exchange', 'Report', 'Security', 'SharePoint', 'Defender', 'Teams', 'PowerPlatform'
+    }
+
+    Update-ModuleManifest @ManifestUpdates
+
+    $CurrentErrorActionPreference = $ErrorActionPreference
+    $ErrorActionPreference = "SilentlyContinue"
+    $Result = Test-ModuleManifest -Path $ManifestPath
+    $ErrorActionPreference = $CurrentErrorActionPreference
```

Comment:
This code confuses me.  It appears to use $ErrorActionPreference (line 156) before it sets its value (line 157).  Is this variable set somewhere else?  If so, might be good to comment where/how that it done.